### ISSUE TITLE
Uploading a file put's it in 2 different folders

### ISSUE THREAD

**Issue by hilite24** [YouTransfer/YouTransfer#180]
I just noticed that somehow when uploading a file, youtransfer "uploads" it a second time to another folder. I noticed this when trying to upload a 3GB file. With the command _df_ the space used went up 6GB instead of only 3GB. After seemingly finishing uploading the file, it doesn't show the finished upload window but waits for a few more seconds. This "wait time" is probably when Youtransfer is copying it to the second folder.

In the youtransfer settings, the local file system path is set to /opt/youtransfer/uploads . Tried it again with a smaller file of 1.7M which gets correctly put it in /opt/youtransfer/uploads :
_-rw-r--r-- 1 root root **1.7M** Mar 16 08:55 37c828d1445fcaa95a0d2ed187d67390.binary
-rw-r--r-- 1 root root  503 Mar 16 08:55 37c828d1445fcaa95a0d2ed187d67390.json_

But somehow also in the youtransfer folder at /opt/youtransfer, just under a different name :
_-rw-r--r--  1 root root **1.7M** Mar 16 08:55 upload_1150d878777d63ed5603e2399ffff545_

Is this a problem with my installation or does anyone notice the same behavior?

Thank you for any kind of help.

**Comment by Syam** [comment]
Hello,

Meet the same issue, sucessfull uploads remain in both `/opt/youtransfer/uploads` and `/opt/youtransfer` as `upload_xxxxxx` tmp file.

It is consumming space twice.

**Comment by remie** [comment]
I can confirm this issue as the current storage providers (localstorage & s3storage) do not clean up the temporary uploaded file after they copied it to the final location. It would have been better to move the file and do the encryption (if enabled) in place instead of streaming it from one location to the other on the file system (and forgetting to remove it).

If anyone feels up to it, the place to fix it is in [localstorage.js](  https://github.com/YouTransfer/YouTransfer/blob/9f059c106e240507ec5766c4d479bcd520ce707d/lib/localstorage.js#L56) and 
[s3storage.js](https://github.com/YouTransfer/YouTransfer/blob/9f059c106e240507ec5766c4d479bcd520ce707d/lib/s3storage.js#L63)

I would be happy to accept a PR that fixes this.

**Comment by alanlitster** [comment]
I managed to fix the issue for the local storage by adding
```javascript
                        fs.unlink(file.path, function(err) {
                                next(err);
                        });
```
to [localstorage.js](https://github.com/YouTransfer/YouTransfer/blob/1.0.6/lib/localstorage.js#L65)

so it then looks like
```javascript
		writeStream.on('finish', function() {
			fs.writeFile(context.jsonPath, JSON.stringify(context), 'utf8', function(err) {
				next(err, context);
			});
                        fs.unlink(file.path, function(err) {
                                next(err);
                        });
		});
```
I'm no javascript programmer but the above fixed it for me.

The changes were made to 1.0.6 which is the latest docker image available and the only one that would work for me.

**Comment by doug86i** [comment]
Is there any progress on this? I like the project allot but will have to find an alternative if this isn't easily fixed. My root partition is a small SSD with a large array of spinning rust as storage, I can't risk the root partition filling up with duplicates.

**Comment by ghost** [comment]
Workaround that worked for me was to create a cronjob to clean the temp folder.

**Comment by doug86i** [comment]
Thanks hilite, I assume it's best to rebuild the docker image to run cron inside the same container? I'm relatively new to docker but enjoying the possibilities!

**Comment by alanlitster** [comment]
@doug86i  No, one container per process. But running cron separately for this is probably an overkill.

You'd be better off creating a script that calls `find` to remove files over a certain age, that way if it happens to run at the same time as a file is being uploaded it won't mess it up.

But as this is only trivial, it'd be easier to just run `docker exec` with your script via your hosts own cron daemon to clean them up. You may want to check that your container is running first before the docker cleanup command.
```
container=youtransfer
 [[ $(docker container ls --no-trunc --filter "name=^/$container$" --filter 'status=running' --format '{{.Names}}') == "$container" ]] && docker exec $container cleanup
```
so that would want to go in it's own script and that would be called via cron on the host.

**Comment by hilite24** [comment]
I'm very new to docker aswell. Sorry for my vague comment.

Here is what I did exactly inside the youtransfer docker container : 

>Create Cleanup script, which finds and deletes files older then 2 months 
 `find /opt/youtransfer/uploads/ -mtime +60 -exec rm {} \;`


>Edit crontab  
   `crontab -e`


>add this line to crontab that executes the script every 12 hours 
`0 */12 * * * exec /root/bin/CleanupYoutransfer.sh`
